bubble_sort.rb

```diff
@@ -0,0 +1,35 @@
+def bubble_sort(array)
+  n = array.length
+  no_swapped = true
+  # this loop code runs until there is nothing to swap
+  while no_swapped
+    no_swapped = false
+    (1...n).each do |i|
+      if array[i - 1] > array[i]
+        array[i - 1], array[i] = array[i], array[i - 1]
+        no_swapped = true
+      end
+    end
+  end
+  p array
+end
+
+def bubble_sort_by(arr)
+  n = arr.length
+  no_swapped = true
+  # swapp loop
+  while no_swapped
+    no_swapped = false
+    (1...n).each do |i|
+      # compare the length of the strings into the array
+      if arr[i - 1].length > arr[i].length
+        arr[i - 1], arr[i] = arr[i], arr[i - 1]
+        no_swapped = true
+      end
+    end
+  end
+  yield p arr
```

Comment:
An error occurred while loading ./bubble_sort_spec.rb.
`Failure/Error: yield p arr`

Kindly check this line of code to fix the issue.